src/compiler/emitter.ts

```diff
@@ -4046,7 +4006,7 @@ module ts {
                 var preambleEmitted = writer.getTextPos() !== outPos;
 
                 if (!preambleEmitted && nodeEndIsOnSameLineAsNodeStart(body, body)) {

```

Comment:
What if there are statements, but the function is still all on one line? Many stubs are written this way:
```ts
var obj = {
    toString: () => { return ""; }
};
```